WebRoot/js/zimbraMail/prefs/model/ZmFilterRules.js

```diff
@@ -376,11 +376,17 @@ function(index, notify, callback, result) {
 ZmFilterRules.prototype._handleErrorSaveRules =
 function(ex) {
 	if (ex.code == ZmCsfeException.SVC_PARSE_ERROR ||
-		ex.code == ZmCsfeException.SVC_INVALID_REQUEST)
+		ex.code == ZmCsfeException.SVC_INVALID_REQUEST ||
+		ex.code == ZmCsfeException.MAIL_NO_SUCH_FOLDER)
```

Comment:
You may also get similar exception if a tag does not exist. Are you handling that?